File: src/main/java/housing/House.java
package housing;

/**************************************************************************************************
 * Class to represent a house with all its intrinsic characteristics.
 *
 * @author daniel, Adrian Carro
 *
 *************************************************************************************************/
public class House implements Comparable<House> {

    //------------------//
    //----- Fields -----//
    //------------------//

    private static int 	id_pool = 0;

    public IHouseOwner  owner;
    public Household    resident;
    public int          id;

    HouseOfferRecord saleRecord;
    HouseOfferRecord rentalRecord;

    private int         quality;

    //------------------------//
    //----- Constructors -----//
    //------------------------//

    /**
     * Creates a house of quality quality in region region
     *
     * @param quality Quality band characterizing the house
     */
	public House(int quality) {
		this.id = ++id_pool;
        this.owner = null;
        this.resident = null;
		this.quality = quality;
	}

    //-------------------//
    //----- Methods -----//
    //-------------------//

	boolean isOnMarket() { return saleRecord != null; }

	HouseOfferRecord getSaleRecord() { return saleRecord; }

	HouseOfferRecord getRentalRecord() { return rentalRecord; }

	boolean isOnRentalMarket() { return rentalRecord != null; }
    void putForSale(HouseOfferRecord saleRecord) { this.saleRecord = saleRecord; }

	void resetSaleRecord() { saleRecord = null; }
    void putForRent(HouseOfferRecord rentalRecord) { this.rentalRecord = rentalRecord; }

	void resetRentalRecord() { rentalRecord = null; }

	public int getQuality() { return quality; }

	@Override
	public int compareTo(House o) { return((int)Math.signum(id-o.id)); }
	
}

File: src/main/java/housing/House.py
import numpy as np

#*************************************************************************************************
# Class to represent a house with all its intrinsic characteristics.
#
# @author daniel, Adrian Carro
#
#************************************************************************************************/
class House:
    id_pool = 0

    # Creates a house of quality quality in region region
    #
    # @param quality Quality band characterizing the house
    def __init__(self, quality: int):
        self.id: int = House.id_pool
        House.id_pool += 1
        self.owner: IHouseOwner = None
        self.resident: Household = None
        self.quality: int = quality

        self.saleRecord: HouseOfferRecord = None
        self.rentalRecord: HouseOfferRecord = None

    #-------------------#
    #----- Methods -----#
    #-------------------#

    def isOnMarket(self) -> bool:
        return self.saleRecord is not None

    def getSaleRecord(self) -> HouseOfferRecord:
        return self.saleRecord

    def getRentalRecord(self) -> HouseOfferRecord:
        return self.rentalRecord

    def isOnRentalMarket(self) -> bool:
        return self.rentalRecord is not None

    def putForSale(self, saleRecord: HouseOfferRecord) -> None:
        self.saleRecord = saleRecord

    def resetSaleRecord(self) -> None:
        self.saleRecord = None

    def putForRent(rentalRecord: HouseOfferRecord) -> None:
        self.rentalRecord = rentalRecord

    def resetRentalRecord(self) -> None:
        self.rentalRecord = None

    def getQuality(self) -> int:
        return self.quality

    def compareTo(House o) -> int:
        return np.sign(self.id - o.id)
